lib/pf/Switch/Cisco.pm

```diff
@@ -1787,6 +1791,64 @@ sub cmnMacChangedNotificationTrapNormalizer {
     return $trapHashRef;
 }
 
+=item acctProfiling
+
+Extract information from radius attributes
+
+=cut
+
+sub acctProfiling {
+    my ( $self, $radius_request) = @_;
+    my $decoded = decode_avpair($radius_request);
+    if (isenabled($self->{_VoIPAccountingDetect})) {
+        $self->acctVoipDetect($decoded);
+     }
+    if (isenabled($self->{_RadiusFingerprint})) {
+        $self->acctFingerprint($decoded);
+     }
+}
+
+=item acctVoipDetect
+
+Extract the cdp/lldp capabilitie from the radius accounting attribute.
+
+=cut
+
+sub acctVoipDetect {
+    my ( $self, $decoded) = @_;
+    $decoded->{'reason'} = 'node_modify';
+
+    if ($decoded->{'isPhone'}) {
+        my $node_attributes = node_attributes($decoded->{mac});
+        if ( $node_attributes->{'voip'} eq 'no') {
+            require pf::role;
+            my $role_obj = new pf::role::custom();
+            my %autoreg_node_defaults = $role_obj->getNodeInfoForAutoReg($decoded);
+            $node_attributes = merge($node_attributes, \%autoreg_node_defaults);
+            my $apiclient = pf::client::getClient;
+            $apiclient->notify('modify_node', %{$node_attributes} );
+            $apiclient->notify('reevaluate_access', %{$decoded} ) if ( $node_attributes->{'voip'} eq 'no');
```

Comment:
constant for no

Reply:
Shouldn't we also set voip to yes after this ?

Reply:
Yes, the cisco_device_sensor will return isPhone to yes if it detect cdp/lldp voip flag.